### PR TITLE
Adds add and remove buttons for toggling properties.

### LINKED ISSUE #956: Remove button for expanded properties
As discussed, the initial toggle of a property involves adding the embedded property to the state, which may have defaults. This ticket is for adding the corresponding remove button.

Note that this does not cover the UX enhancements that Astrid is working on, just adding the remove functionality.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #956

### CODE REVIEW

File: __tests__/components/editor/property/OutlineHeader.test.js [hunk 1]
```diff
@@ -3,44 +3,141 @@
...
     propertyURI: 'http://id.loc.gov/ontologies/bibframe/instanceOf',
     mandatory: 'false',
   }
-  const headerProps = {
-    spacer: 0,
-    collapsed: true,
-    pt: property,
-  }
-  const wrapper = shallow(<OutlineHeader {...headerProps} />)
+  // const mockEvent = { preventDefault: () => {} }
```
Comment (jcoyne) [inline __tests__/components/editor/property/OutlineHeader.test.js]: Cruft?

File: src/components/editor/property/OutlineHeader.jsx [hunk 2]
```diff
@@ -40,10 +42,41 @@ OutlineHeader.propTypes = {
...
+
+  return {
+    resourceModel,
+    property,
+    collapsed: !isExpanded(state.selectorReducer, ourProps.reduxPath),
+  }
+}
+
+const mapDispatchToProps = (dispatch, ownProps) => ({
+  handleTogglePlusButton: (event) => {
```
Comment (jcoyne) [inline src/components/editor/property/OutlineHeader.jsx]: As this is not a "plus" button anymore, can you please change the name to "handleToggleCollapse" or something without "plus" in it.

File: src/components/editor/property/PropertyTemplateOutline.jsx [hunk 3]
```diff
@@ -60,7 +54,7 @@ PropertyTemplateOutline.propTypes = {
   property: PropTypes.object.isRequired,
   resourceModel: PropTypes.object,
   collapsed: PropTypes.bool,
-  handleTogglePlusButton: PropTypes.func,
+  // handleTogglePlusButton: PropTypes.func,
```
Comment (jcoyne) [inline src/components/editor/property/PropertyTemplateOutline.jsx]: Cruft?

File: src/reducers/index.js [hunk 4]
```diff
@@ -54,8 +54,8 @@ export const appendResource = (state, action) => {
 export const removeResource = (state, action) => {
   const newState = _.cloneDeep(state)
   const reduxPath = action.payload
-  const key = reduxPath.slice(-2)[0]
-  const parentReduxPath = reduxPath.slice(0, reduxPath.length - 2)
+  const key = reduxPath.slice(-1)[0]
```
Comment (jcoyne) [inline src/reducers/index.js]: It's odd that this didn't require a change to the test for removeResource.   Do we have a good test there?

File: __tests__/components/editor/property/OutlineHeader.test.js [hunk 5]
```diff
@@ -3,44 +3,141 @@
...
+    })
 
-    expect(faWrapper.getElement(0).props.icon).toEqual(faMinusSquare)
+    it('has a remove button', () => {
+      expect(wrapper.exists('button.btn-remove')).toEqual(true)
+    })
   })
+
+
+  // it('contains a FontAwesomeIcon', () => {
```
Comment (jcoyne) [inline __tests__/components/editor/property/OutlineHeader.test.js]: Cruft?

File: src/reducers/index.js (hunk 4 above)
Comment (justinlittman) [reply src/reducers/index.js]: See the change in __tests__/reducers/index.test.js

**Comment by justinlittman** [comment]
@jcoyne Removed cruft and renamed function.

File: src/components/editor/property/OutlineHeader.jsx [hunk 6]
```diff
@@ -40,10 +42,41 @@ OutlineHeader.propTypes = {
...
   handleCollapsed: PropTypes.func,
   id: PropTypes.string,
-  isRequired: PropTypes.any,
-  isAdd: PropTypes.bool,
-  pt: PropTypes.object,
-  spacer: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  property: PropTypes.object,
+  reduxPath: PropTypes.array,
+  resourceModel: PropTypes.object,
+  handleToggleAddButton: PropTypes.func,
```
Comment (jcoyne) [inline src/components/editor/property/OutlineHeader.jsx]: Why is it `handleToggleAddButton` rather than `handleAddButton`?  Is it toggleable?

Comment (justinlittman) [reply src/components/editor/property/OutlineHeader.jsx]: It actually handles 2 buttons: the toggle (plus) and the add button (which is add and then toggle).

Comment (jcoyne) [reply src/components/editor/property/OutlineHeader.jsx]: To me toggle means open and close.  So, `handleAddAndOpen` is preferable in that case.  I think we could also call this `handleAdd` as that is the users intent.  The fact that "add" has two actions associated with it is the business of a the reducer.

Comment (justinlittman) [reply src/components/editor/property/OutlineHeader.jsx]: Renamed.

**Pull request merged**